version.go

```diff
@@ -10,13 +10,17 @@ import (
 // versionCommand is a cmd.Command that prints the current version.
 type versionCommand struct {
 	CommandBase
-	out     Output
-	version string
+	out           Output
+	version       string
+	versionDetail interface{}
```

Comment:
I feel like we should be specific about the details that we want to expect. Should we be creating a specific interface with information that we're hoping to be able to pass on?

Reply:
My thought was to leave it up to each individual supercommand as they could have different expectations of what version information to pass. The version string should be the bare minimum, but if a command wants to include say Build date, this allows them to add it without imposing it on all users of juju/cmd.

Reply:
I like the idea of interface{} here because it provides more flexibility and is less imposing. I like that it is not more rigidly structured.